src/typechecker.erl

```diff
@@ -4155,9 +4181,9 @@ handle_type_error({type_error, map, P, ResTy, MapTy}) ->
 handle_type_error({type_error, mismatch, Ty, Expr}) ->
     io:format("The expression ~s at line ~p does not have type ~s~n",
               [erl_pp:expr(Expr), erl_anno:line(element(2, Expr)), typelib:pp_type(Ty)]);
-handle_type_error({bad_type_annotation, P, TypeStr}) ->
+handle_type_error({bad_type_annotation, TypeLit}) ->
     io:format("The type annotation ~p on line ~p is not a valid type~n",
-              [TypeStr, erl_anno:line(P)]);
+              [erl_pp:expr(TypeLit), line_no(TypeLit)]);
```

Comment:
you could add `maybe_format_column` too

Reply:
Maybe. I thought it can be done when column is added to all error messages in a uniform way. One function call `format_location` would be preferable.